cip/CIP2016-06-14-Define-comparability-and-equality-as-well-as-orderability-and-equivalence.adoc

```diff
@@ -228,39 +233,46 @@ To accomplish this, there must be a pre-determined order of types and each value
 
 The accompanying descending global sort order is the same order in reverse (i.e. it runs from `VOID` to `MAP`).
 
 [[equivalence-def,equivalence]]
 ==== Equivalence ====
 
-Equivalence remains mostly unchanged but now can be defined succinctly as being identical to equality except that:
+Equivalence now can be defined succinctly as being identical to equality except that:
 
 * Any two `null` values are equivalent (both directly or inside nested structures).
 * Any two `NaN` values are equivalent (both directly or inside nested structures).
 * However, `null` and `NaN` values are not equivalent (both directly or inside nested structures).
+* Equivalence of lists is identical to equality of lists but uses equivalence for comparing the contained list elements.
+* Equivalence of regular maps is identical to equality of regular maps but uses equivalence for comparing the contained map entries.
 
 Equivalence is reflexive for all values.
 
 [[aggregation]]
 === Aggregation ===
 
-Generally, an aggregation function call `aggr(expr)` processes all matching rows for each aggregation key found in an incoming record (keys are compared using equivalence, as noted above).
+Generally an aggregation `aggr(expr)` processes all matching rows for each aggregation key found in an incoming record (keys are compared using equivalence).
+
 For a fixed aggregation key and each matching record, `expr` is evaluated to a value.
 This yields a list of candidate values.
-Generally, the order of candidate values is unspecified.
-However, if the aggregation happens in a projection with an associated `ORDER BY` subclause, the list of candidate values is ordered in the same way as the underlying records and as specified by the associated `ORDER BY` subclause.
+Generally the order of candidate values is unspecified.
+If the aggregation happens in a projection with an associated `ORDER BY` subclause, the list of candidate values is ordered in the same way as the underlying records and as specified by the associated `ORDER BY` subclause.
 
 In a regular aggregation (i.e. of the form `aggr(expr)`), the list of aggregated values is the list of candidate values with all `null` values removed from it.
 
 In a distinct aggregation (i.e. of the form `aggr(DISTINCT expr)`), the list of aggregated values is the list of candidate values with all `null` values removed from it.
-Futhermore, in a distinct aggregation, only the first of all equivalent candidate values is included in the list of aggregated values, i.e. duplicates under equivalence are removed.
+Futhermore, in a distinct aggregation, only one of all equivalent candidate values is included in the list of aggregated values, i.e. duplicates under equivalence are removed.
+However, if the distinct aggregation happens in a projection with an associated `ORDER BY` subclause, always the first from a set of equivalent candidate values is included in the list of aggregated values.
```

Comment:
Shouldn't it be `only one element from each set of equivalent candidate values ...`?